### PR TITLE
Ensure whos online widget and page use same timing calculation

### PR DESCRIPTION
**Pull request by scottcwilson**
who's online results in dashboard widget differ from who's online page.

### CODE REVIEW

**Comment by drbyte** [comment]
NOTE: it is helpful when both the "commit title" (ie: first line of text in your actual commit message) and your PR title contain context-relevant descriptions.
In this case there's no context that this has something to do with the Admin, nor that it is specific to Whos-online.

More useful (again, for both the commit and the PR title) would be something along the lines of 
`Update whos-online limit display in admin dashboard`

**Comment by scottcwilson** [comment]
PLEASE HOLD on this PR.  More fixes coming in this area.

**Comment by scottcwilson** [comment]
Issue symptom: dashboard would show pink for a customer where the page would show red.

**Comment by drbyte** [comment]
> Issue symptom: dashboard would show pink for a customer where the page would show red.

?

**Comment by scottcwilson** [comment]
Open admin dashboard in one tab, admin whos online in a second, and browser in a third.  Who's online will show browser session as red, but dashboard will show it as pink.  (Assumption is you are not filtering out Admin IP Addresses.)

**Comment by drbyte** [comment]
I'm going to continue to be pedantic, to make my point:

> Open admin dashboard in one tab, admin whos online in a second, and browser in a third. Who's online will show browser session as red, but dashboard will show it as pink.

... and this can be boiled down to a 60-character description which explains the original problem in a meaningful way to someone looking at it 6 months from now, by restating it as ... ________ ?
(both in the PR title and in the commit title (force-pushed to rewrite the git log correctly))

**Comment by scottcwilson** [comment]
These were repro instructions, not a brief summary.  I thought that's what you wanted.  If you can be a bit more explicit than "?" I might be able to get it right faster.

File: admin/includes/languages/english/whos_online.php [hunk 1]
```diff
@@ -40,7 +40,8 @@
 define('TEXT_WHOS_ONLINE_FILTER_ADMINS', 'Exclude Admin IP Addresses?');
 
 define('WHOIS_TIMER_REMOVE', 1200); // seconds when removed from whos_online table - 1200 default = 20 minutes
-define('WHOIS_TIMER_INACTIVE', 180); // seconds when considered inactive - 180 default = 3 minutes
+// moved to english.php 
+// define('WHOIS_TIMER_INACTIVE', 180); // seconds when considered inactive - 180 default = 3 minutes
```
Comment (drbyte) [inline admin/includes/languages/english/whos_online.php:44]: Not loving this change. Breaks up the context.

Sure, to use the same constant elsewhere it must be shared, but that makes me wonder how important this even is ...

Comment (scottcwilson) [reply admin/includes/languages/english/whos_online.php:44]: Could move all 3 to english.php Without this, the 2 pages are out of sync - wouldn't you call that a confusing bug?  One of my customers complained as soon as I put 1.5.6 on their site.

**Comment by drbyte** [comment]
Title is better. Thanks.

**Comment by drbyte** [comment]
The changes you've made here are multifold:
- changed queries to avoid any caching, which slows things down
- changed time threshold on dashboard
- changed location of constant to couple the dashboard with the whosonline page

I'm guessing the root cause here is that you'd changed the threshold in the constant but then you wanted the dashboard to use the same number? 

Or was the root cause that the queries were cached and you didn't like that?

File: admin/includes/languages/english/whos_online.php (hunk 1 above)
Comment (drbyte) [reply admin/includes/languages/english/whos_online.php:44]: > Could move all 3 to english.php

Sigh. Yes we certainly could. 

But then everyone's gonna complain that these are not language-related at all, and complain that we're cluttering english.php, and want more configuration switches that are already too cumbersome ...

**Comment by scottcwilson** [comment]
Root cause was dashboard should use same number. 
Noticed the queries were cached and thought this could also be misleading.

File: admin/includes/languages/english/whos_online.php (hunk 1 above)
Comment (scottcwilson) [reply admin/includes/languages/english/whos_online.php:44]: extra_configures?  Put in db?  What would you like to do, Chris?

**Comment by scottcwilson** [comment]
Ugh still seeing wildly different numbers on dashboard vs whos online page.  Have you tested this on a busy store?

**Comment by drbyte** [comment]
Questions:
- in what ways do people commonly customize the admin whos-online page? 
- are there any mods that alter it? In what way?

Or is it largely untouched by everyone?

**Comment by scottcwilson** [comment]
I would guess it is untouched.

**Comment by wiztechinc** [comment]
The only thing I’ve ever done comes with the cowoa mod. Though I actually wish some of it would go away.  So if I could, I would pick and choose what I wanted my clients to see.

**Comment by drbyte** [comment]
> The only thing I’ve ever done comes with the cowoa mod. Though I actually wish some of it would go away. So if I could, I would pick and choose what I wanted my clients to see.

Cool - that's useful: Do you mind elaborating about what you'd like to hide/show etc?

**Comment by wiztechinc** [comment]
The only thing I personally ever look at is new customers and new orders. The ones I would like to be able to hide from my clients are stats, who's online (I actually detest it in general because I've had clients who get hung up on watching it and obsessing over each entry.), visitor history, orders at the bottom left is barely noticeable down there so some might want that to be more prominent. specials - some never run specials. 

In other words each cart owner is different, including me. The idea of being able to move the widgets or turn them off was one of the greatest ideas I've heard and I'm very disappointed this hasn't been implimented.

**Comment by drbyte** [comment]
Okay. Thanks for clarifying that you only meant you prefer to remove it because you think people misuse it.

Unfortunately that's not what I was asking.

**Comment by drbyte** [comment]
Aside: I looked into the various implementations of COWOA and I don't see any ways in which it alters whosonline operation. If I'm missing something in that regard, please show me.

**Comment by wiztechinc** [comment]
> Aside: I looked into the various implementations of COWOA and I don't see any ways in which it alters whosonline operation. If I'm missing something in that regard, please show me.

nope it doesn't - I was talking about changes to the dashboard - and not who's online. Sorry! I guess I wasn't following as closely as I thought I was. 

But I don't feel like it's something that should be on the dashboard at all.

**Comment by zcwilt** [comment]
I guess this is where people could complain about our lack of a `roadmap`
However we did move to using a new system for dashboard widgets in the original v2 branch and 
that would still be what we want to move to. Given that I'd like to defer this to v158 and until we can think about roadmap again.

**Comment by scottcwilson** [comment]
My vote would be to remove this from the default dashboard entirely and close the issue.  The presence of this widget creates a "two sources of truth" problem which drives nervous shopowners crazy as the obsess over the differences between these dashboard and the whos online page.

**Comment by drbyte** [comment]
While I agree that we can improve the dashboard to make it easier to turn things off,
this is the first time I've heard anyone complain about having the widget on the dashboard. Are there a bunch of forum discussions about it?

**Comment by wiztechinc** [comment]
probably not - it's the developers who pay the price
